src/libraries/Microsoft.Extensions.Hosting.Systemd/src/SystemdHostBuilderExtensions.cs

```diff
@@ -27,27 +27,63 @@ public static class SystemdHostBuilderExtensions
         ///     notifications. See https://www.freedesktop.org/software/systemd/man/systemd.service.html.
         ///   </para>
         /// </remarks>
-        /// <param name="hostBuilder">The <see cref="IHostBuilder"/> to use.</param>
-        /// <returns></returns>
+        /// <param name="hostBuilder">The <see cref="IHostBuilder"/> to configure.</param>
+        /// <returns>The <paramref name="hostBuilder"/> instance for chaining.</returns>
         public static IHostBuilder UseSystemd(this IHostBuilder hostBuilder)
         {
             if (SystemdHelpers.IsSystemdService())
             {
                 hostBuilder.ConfigureServices((hostContext, services) =>
                 {
-                    services.Configure<ConsoleLoggerOptions>(options =>
-                    {
-                        options.FormatterName = ConsoleFormatterNames.Systemd;
-                    });
-
-                    // IsSystemdService() will never return true for android/browser/iOS/tvOS
-#pragma warning disable CA1416 // Validate platform compatibility
-                    services.AddSingleton<ISystemdNotifier, SystemdNotifier>();
-                    services.AddSingleton<IHostLifetime, SystemdLifetime>();
-#pragma warning restore CA1416 // Validate platform compatibility
+                    AddSystemdLifetime(services);
                 });
             }
             return hostBuilder;
         }
+
+        /// <summary>
+        /// Configures the lifetime of the <see cref="IHost"/> built from <paramref name="services"/> to
+        /// <see cref="SystemdLifetime"/>, provides notification messages for application started
+        /// and stopping, and configures console logging to the systemd format.
+        /// </summary>
+        /// <remarks>
+        ///   <para>
+        ///     This is context aware and will only activate if it detects the process is running
+        ///     as a systemd Service.
+        ///   </para>
+        ///   <para>
+        ///     The systemd service file must be configured with <c>Type=notify</c> to enable
+        ///     notifications. See <see href="https://www.freedesktop.org/software/systemd/man/systemd.service.html"/>.
+        ///   </para>
+        /// </remarks>
+        /// <param name="services">
+        /// The <see cref="IServiceCollection"/> used to build the <see cref="IHost"/>.
+        /// For example, <see cref="HostApplicationBuilder.Services"/> or the <see cref="IServiceCollection"/> passed to the
+        /// <see cref="IHostBuilder.ConfigureServices(System.Action{HostBuilderContext, IServiceCollection})"/> callback.
+        /// </param>
+        /// <returns>The <paramref name="services"/> instance for chaining.</returns>
+        public static IServiceCollection AddSystemd(this IServiceCollection services)
+        {
```

Comment:
In extensions, so we typically null check receivers to throw an ArgumentNullException instead of null ref'ing? If no, this is fine. If yes, we should here, too.

Reply:
It's pretty inconsistent. Sometimes we don't check:

https://github.com/dotnet/runtime/blob/807d0916c5205d75ecb04809f8106a3caa46bbe4/src/libraries/Microsoft.Extensions.Hosting/src/HostingHostBuilderExtensions.cs#L30-L86

https://github.com/dotnet/runtime/blob/807d0916c5205d75ecb04809f8106a3caa46bbe4/src/libraries/Microsoft.Extensions.Logging/src/LoggingBuilderExtensions.cs#L22-L62

But for all `IServiceCollection` args we do seem to check:

https://github.com/dotnet/runtime/blob/807d0916c5205d75ecb04809f8106a3caa46bbe4/src/libraries/Microsoft.Extensions.Logging/src/LoggingServiceCollectionExtensions.cs#L32-L34

https://github.com/dotnet/runtime/blob/807d0916c5205d75ecb04809f8106a3caa46bbe4/src/libraries/Microsoft.Extensions.Options.ConfigurationExtensions/src/OptionsConfigurationServiceCollectionExtensions.cs#L62-L65

So I'll add them here.